frontend/src/metabase/admin/settings/components/widgets/SettingInput/SettingInput.tsx

```diff
@@ -10,33 +11,41 @@ const getValue = (value: string, type: string) => {
   return value;
 };
 
-interface SettingInputProps {
+type Value = string | number | null;
+
+export interface SettingInputProps {
   setting: {
     key: string;
     value: string | null;
     default?: string;
     placeholder?: string;
   };
-  onChange: (value: string | number | null) => void;
+  onChange: (value: Value) => void;
   autoFocus?: boolean;
   fireOnChange?: boolean;
   errorMessage?: string;
   id?: string;
   type?: string;
+  normalize?: (value: Value, props: Partial<SettingInputProps>) => Value;
 }
 
-const SettingInput = ({
+export const SettingInput = ({
   setting,
   onChange,
   autoFocus,
   errorMessage,
   fireOnChange,
   id,
   type = "text",
+  normalize = value => value,
 }: SettingInputProps) => {
+  const normalizeWithProps = useCallback(
+    (value: Value) => normalize(value, { type }),
```

Comment:
if we remove the `type` check in the other comment, this could be removed

Reply:
Replied here https://github.com/metabase/metabase/pull/36693#discussion_r1429862552. 

I don't think we can remove it.

Let me know what you think.

Reply:
Fixed in https://github.com/metabase/metabase/pull/36693/commits/8a26f5420498ad4a8992ebcc21bb6465e3fe0b01